### ISSUE TITLE
array with items field in oneOf

### ISSUE THREAD

**Issue by aurelien-brevet** [rjsf-team/react-jsonschema-form#3877]
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)
- [X] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net), [Codesandbox.io](https://codesandbox.io) or preferably a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

core

### Version

5.13.0

### Current Behavior

Error: Unsupported field schema for field root_possibleValues: Missing items definition.

### Expected Behavior

No error.
Array with either elements of type number or string.

### Steps To Reproduce

```
"possibleValues": {
      "type": "array",
      "oneOf": [
        {
          "items": {
            "type": "number"
          }
        },
        {
          "items": {
            "type": "string"
          }
        }
      ]
}
```

### Environment

```markdown
- OS:
- Node:
- npm:
```

**Comment by heath-freenome** [comment]
Link to the [playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJwb3NzaWJsZVZhbHVlcyI6W119LCJzY2hlbWEiOnsidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsicG9zc2libGVWYWx1ZXMiOnsidHlwZSI6ImFycmF5Iiwib25lT2YiOlt7Iml0ZW1zIjp7InR5cGUiOiJudW1iZXIifX0seyJpdGVtcyI6eyJ0eXBlIjoic3RyaW5nIn19XX19fSwidWlTY2hlbWEiOnt9LCJ0aGVtZSI6ImRlZmF1bHQiLCJsaXZlU2V0dGluZ3MiOnsic2hvd0Vycm9yTGlzdCI6InRvcCIsInZhbGlkYXRlIjpmYWxzZSwiZGlzYWJsZWQiOmZhbHNlLCJub0h0bWw1VmFsaWRhdGUiOmZhbHNlLCJyZWFkb25seSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOmZhbHNlLCJsaXZlT21pdCI6ZmFsc2UsImV4cGVyaW1lbnRhbF9kZWZhdWx0Rm9ybVN0YXRlQmVoYXZpb3IiOnsiYXJyYXlNaW5JdGVtcyI6InBvcHVsYXRlIiwiZW1wdHlPYmplY3RGaWVsZHMiOiJwb3B1bGF0ZUFsbERlZmF1bHRzIn19fQ==) with this error:

**Comment by heath-freenome** [comment]
@aurelien-brevet I switched around the JSON Schema a little bit in this [playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJwb3NzaWJsZVZhbHVlcyI6W119LCJzY2hlbWEiOnsidHlwZSI6Im9iamVjdCIsInByb3BlcnRpZXMiOnsicG9zc2libGVWYWx1ZXMiOnsidHlwZSI6ImFycmF5IiwiaXRlbXMiOnsib25lT2YiOlt7InRpdGxlIjoiTnVtYmVyIiwidHlwZSI6Im51bWJlciJ9LHsidGl0bGUiOiJTdHJpbmciLCJ0eXBlIjoic3RyaW5nIn1dfX19fSwidWlTY2hlbWEiOnt9LCJ0aGVtZSI6ImRlZmF1bHQiLCJsaXZlU2V0dGluZ3MiOnsic2hvd0Vycm9yTGlzdCI6InRvcCIsInZhbGlkYXRlIjpmYWxzZSwiZGlzYWJsZWQiOmZhbHNlLCJub0h0bWw1VmFsaWRhdGUiOmZhbHNlLCJyZWFkb25seSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOmZhbHNlLCJsaXZlT21pdCI6ZmFsc2UsImV4cGVyaW1lbnRhbF9kZWZhdWx0Rm9ybVN0YXRlQmVoYXZpb3IiOnsiYXJyYXlNaW5JdGVtcyI6InBvcHVsYXRlIiwiZW1wdHlPYmplY3RGaWVsZHMiOiJwb3B1bGF0ZUFsbERlZmF1bHRzIn19fQ==) and am wondering if this works for you?

**Comment by aurelien-brevet** [comment]
HI @heath-freenome, Thanks for your suggestion, but the result is not the same. I want to prohibit mixing values of different types.

Using my schema:

```
{
     "possibleValues": [1, "string"]
}
```

is not valid but valid with yours.

**Comment by heath-freenome** [comment]
@aurelien-brevet Good point. I believe that this [playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6e30sInNjaGVtYSI6eyJ0eXBlIjoib2JqZWN0Iiwib25lT2YiOlt7InRpdGxlIjoiTnVtYmVycyIsInByb3BlcnRpZXMiOnsicG9zc2libGVWYWx1ZXMiOnsidHlwZSI6ImFycmF5IiwiaXRlbXMiOnsidHlwZSI6Im51bWJlciJ9fX19LHsidGl0bGUiOiJTdHJpbmdzIiwicHJvcGVydGllcyI6eyJwb3NzaWJsZVZhbHVlcyI6eyJ0eXBlIjoiYXJyYXkiLCJpdGVtcyI6eyJ0eXBlIjoic3RyaW5nIn19fX1dfSwidWlTY2hlbWEiOnt9LCJ0aGVtZSI6ImRlZmF1bHQiLCJsaXZlU2V0dGluZ3MiOnsic2hvd0Vycm9yTGlzdCI6InRvcCIsInZhbGlkYXRlIjpmYWxzZSwiZGlzYWJsZWQiOmZhbHNlLCJub0h0bWw1VmFsaWRhdGUiOmZhbHNlLCJyZWFkb25seSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOmZhbHNlLCJsaXZlT21pdCI6ZmFsc2UsImV4cGVyaW1lbnRhbF9kZWZhdWx0Rm9ybVN0YXRlQmVoYXZpb3IiOnsiYXJyYXlNaW5JdGVtcyI6InBvcHVsYXRlIiwiZW1wdHlPYmplY3RGaWVsZHMiOiJwb3B1bGF0ZUFsbERlZmF1bHRzIn19fQ==) example does what you want

**Comment by rbavery** [comment]
I'm also encountering this issue with a number of JSON schemas in the [STAC ecoystem](https://github.com/radiantearth/stac-spec)


When I paste in [this JSON Schema](https://github.com/crim-ca/mlm-extension/blob/main/json-schema/schema.json), it parses a few fields but then errors with

```
Unsupported field schema for field root_stac_extensions: Missing items definition.
```

the section with the missing items definition

```
{
  "type": "array",
  "contains": {
    "const": "https://crim-ca.github.io/mlm-extension/v1.2.0/schema.json"
  }
}
```

I had some trouble figuring out the source of this issue since I'm not too familiar with JS. So I tested out another extension, [the EO extension JSON Schema](https://github.com/stac-extensions/eo/blob/release/v1.1.0/json-schema/schema.json), and it also has this same issue with the same field and error.

When the schema is edited to add an items definition to the stac_extensions field, rendering the EO schame.json works in the form filler playground. However I'm hoping that there can be a way to not require updates to all the schemas for these to work in this tool.

```
"stac_extensions": {
  "type": "object",
  "required": [
    "stac_extensions"
  ],
  "properties": {
    "stac_extensions": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "contains": {
        "const": "https://stac-extensions.github.io/eo/v1.1.0/schema.json"
      }
    }
  }
},
```
